release-content/0.16/release-notes/decals.md

```diff
@@ -0,0 +1,29 @@
+**Decals** are textures which can be dynamically layered on top of existing meshes, conforming to their geometry.
+This has two benefits over simply changing a mesh's texture:
+
+1. You can add them dynamically in response to player actions. Most famously, bullet holes in FPS games use decals for this.
+2. You don't need to create an entirely new texture for every combination, which makes them more efficient and flexible when creating levels with details like graffiti or cracks in building facades.
+
+Like many things in rendering, there are a huge number of ways to implement this feature, each with their own tradeoffs.
+In Bevy 0.16, we've selected two complementary approaches: **forward decals** and **clustered decals**.
+
+TODO: add decal image.
+
+Our implementation of forward decals (or to be more precise, contrast projective decals) was inspired by [Alexander Sannikovs talk on the rendering techniques of Path of Exile 2], and was upstreamed from the [`bevy_contact_projective_decals`] ecosystem crate.
+Due to nature of this technique, looking at the decal from very steep angles will cause distortion.
+This can be mitigated by creating textures that are bigger than the effect, giving the decal more space to stretch.
+To create a forward decal, spawn a [`ForwardDecal`] entity, which uses a [`ForwardDecalMaterial`] using the [`ForwardDecalMaterialExt`] material extension.
+
+Clustered decals (or decal projectors) work by projecting images from a 1x1x1 cube onto surfaces found in the +Z direction.
+They are clusterable objects, just like point lights and light probes, which means that decals are only evaluated for objects within the bounds of the projector, and they don't require a second rendering pass.
```

Comment:
> and they don't require a second rendering pass.

Kind of implies that forward decals _do_ require a second rendering pass, which sorta isn't true. 

Small summary of how they work:
1. Forward decals basically map to mesh + material under the hood, I just did some fancy API stuff to make it ergonomic for users to spawn. It's literately just an extra alpha-blended mesh that gets rendered in the transparent phase.
2. Clustered decals get binned into froxels(iirc?), and then during the opaque pass each fragments lookup what froxel they're in, and then iterate over the decals in the froxel and apply them. Same as how points/spot lights work, and I guess light probes although I forgot about that bit.

Reply:
So forward decals results in the same fragment being shaded twice? In which case:
```suggestion
They are clusterable objects, just like point lights and light probes, which means that decals are only evaluated for objects within the bounds of the projector, and they don't require shading the same fragment an additional time to blend the decal on top.
```